hetmatpy/pipeline.py

```diff
@@ -26,12 +40,38 @@ def add_gamma_hurdle_to_dgp_df(dgp_df):
         )
     # Compute gamma-hurdle parameters
     dgp_df['mean_nz'] = dgp_df['sum'] / dgp_df['nnz']
-    dgp_df['sd_nz'] = ((dgp_df['sum_of_squares'] - dgp_df['sum'] ** 2 / dgp_df['nnz']) / (dgp_df['nnz'] - 1)) ** 0.5
-    dgp_df['beta'] = dgp_df['mean_nz'] / dgp_df['sd_nz'] ** 2
-    dgp_df['alpha'] = dgp_df['mean_nz'] * dgp_df['beta']
+    dgp_df['sd_nz'] = calculate_sd(dgp_df['sum_of_squares'], dgp_df['sum'], dgp_df['nnz'])
+
+    # If the standard deviation is zero, we'll go ahead and set beta and alpha to -1.
+    # This has the benefit of both not dividing by zero and ensuring that the gamma
+    # function breaks if it is still called somehow
+    if dgp_df['sd_nz'] == 0:
+        dgp_df['beta'] = -1
+        dgp_df['alpha'] = -1
+    else:
+        dgp_df['beta'] = dgp_df['mean_nz'] / dgp_df['sd_nz'] ** 2
+        dgp_df['alpha'] = dgp_df['mean_nz'] * dgp_df['beta']
     return dgp_df
 
 
+def calculate_p_value(row, dgp_df):
+    """
+    Calculate the p_value for a given metapath if possible
+    """
+    if row['sum'] == 0:
+        return None
+    elif row['path_count'] == 0:
+        return 1.0
+    # If the standard deviation is zero, calculate the p_value empiricly
+    elif row['sd_nz'] == 0:
+        if row['dwpc'] <= dgp_df['mean_nz']:
+            return dgp_df['nnz'] / dgp_df['n_dwpcs']
```

Comment:
I'm pretty sure these are the two correct fields for calculating the p_value, but I'd appreciate someone double checking it

Reply:
Yes correct.